### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by calebchris000**
# Highlights 🌟 

### In this PR:
- Add the following validations for
  - The User
  - The Post
- Add unit specs for all of your models' methods and validations.

**Peace.** ✌️

### CODE REVIEW

File: app/models/author.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@ class Author < ApplicationRecord
   has_many :posts, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'author_id'
+  validate :name, presence: true
+  validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def last_3_post
     post.order(created_at: :desc).limit(3)
```
Comment (descholar-ceo) [inline app/models/author.rb]: - [ ] Any custom methods in the models should have a corresponding unit test.

File: app/models/post.rb [hunk 2]
```diff
@@ -4,9 +4,9 @@ class Post < ApplicationRecord
...
   belongs_to :author
   has_many :likes
-  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :comment_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   validates :title, presence: true, length: { maximum: 250 }
+  validates :comment_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def top_5_comment
     comments.order(created_at: :desc).limit(5)
```
Comment (descholar-ceo) [inline app/models/post.rb:15]: - [ ] Any custom methods in the models should have a corresponding unit test.

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @calebchris000 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
👍🏽 Linters are passing ✔️ 
👍🏽 Readme file is following the microverse's template ✔️ 
👍🏽 The PR is following the correct git flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

- [ ] The following methods are missing test cases:
- [ ] [last_3_posts](https://github.com/calebchris000/Blog-Application/blob/7550f9f46e7c18038c37b01e184a916c382951bc/app/models/author.rb#L8-L10) from the `Author` model
- [ ] [top_5_comments](https://github.com/calebchris000/Blog-Application/blob/7550f9f46e7c18038c37b01e184a916c382951bc/app/models/post.rb#L11-L14) from `Post` model

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Installing github actions for running your test cases on github whenever you push

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @calebchris000 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**